pkg/config/middleware.go

```diff
@@ -87,3 +94,18 @@ func (a *auth) ServeHTTP(w http.ResponseWriter, req *http.Request, next http.Han
 	}
 	a.JWTMiddleware.HandlerWithNext(w, req, next)
 }
+
+type statsdMiddleware struct {
+	StatsdClient *statsd.StatsdClient
+}
+
+func (s *statsdMiddleware) ServeHTTP(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	defer func(start time.Time) {
+		response := w.(negroni.ResponseWriter)
+		status := strconv.Itoa(response.Status())
```

Comment:
You are aggregating all requests. Maybe you can extend the scope and add the route path to the stats. For example, getting the path from `r.URL.Path` or `r.URL.RawPath`, and add path to the statsd string.

```
path := r.URL.Path
// e.g. /api/v1/evaluation, you may want to replace '/' with something else for safe escapes.

s.StatsdClient.Incr(fmt.Sprintf("http.requests.%s.%s.count", path, status), 1)
s.StatsdClient.PrecisionTiming(fmt.Sprintf("http.requests.%s.duration", path), time.Since(start)
```

Or even better, use a statsd client that supports `tags`. It's nice to just tag the count and duration with path.

https://godoc.org/gopkg.in/alexcesaro/statsd.v2

Reply:
@zhouzhuojie I've been experimenting, the Datadog statsd supports tags, but this one I've used doesn't.

I might switch to the Datadog one, considering its backwards compatible with any other statsd whilst the normal statsd ones have missing features comparing to the Datadog one.

Reply:
in regards to the routes itself, the biggest issue if we include on the metric name is that it gets tricky to remove the "ids", which then becomes a bit useless.

you don't want to track metrics for resources, yet, routes:

`/api/foo/bar/1`  has to become `/api/foo/bar/` to be useful.